cmd/zc_traverser_blob.go

```diff
@@ -418,7 +431,12 @@ func (t *blobTraverser) serialList(containerURL azblob.ContainerURL, containerNa
 	return nil
 }
 
-func newBlobTraverser(rawURL *url.URL, p pipeline.Pipeline, ctx context.Context, recursive, includeDirectoryStubs bool, incrementEnumerationCounter enumerationCounterFunc, s2sPreserveSourceTags bool, cpkOptions common.CpkOptions) (t *blobTraverser) {
+func newBlobTraverser(rawURL *url.URL, p pipeline.Pipeline, ctx context.Context, recursive, includeDirectoryStubs bool, incrementEnumerationCounter enumerationCounterFunc, s2sPreserveSourceTags bool, cpkOptions common.CpkOptions, includeDeleted, includeSnapshot, includeVersion bool) (t *blobTraverser) {
+	if strings.ToLower(glcm.GetEnvironmentVariable(common.EEnvironmentVariable.DisableHierarchicalScanning())) == "false" &&
+		includeDeleted && (includeSnapshot || includeVersion) {
+		os.Setenv("AZCOPY_DISABLE_HIERARCHICAL_SCAN", "true")
```

Comment:
Will this setting gets enabled right away? In case of tests, we set the time.sleep(30) for this, right?

Reply:
Yes, this gets enabled right away. I've tested this locally and it'll enable it immediately.